src/nr_openai_observability/bedrock.py

```diff
@@ -174,6 +185,34 @@ def handle_bedrock_create_completion(response, contents, time_delta, **kwargs):
         raise error
 
 
+def handle_bedrock_embedding(result, contents, time_delta, **kwargs):
+    embedding_id = str(uuid.uuid4())
+    response_body = json.loads(contents)
+    input_body = json.loads(kwargs.get("body"))
+    request_id = None
+    if None != result.get("ResponseMetadata") and None != result.get("ResponseMetadata").get("RequestId"):
+        request_id = result.get("ResponseMetadata").get("RequestId")
+
+    embedding = {
+        "id": embedding_id,
+        "request_id": request_id,
+        "input": input_body.get("inputText")[:4095],
+        "timestamp": datetime.now(),
+        "request.model": kwargs.get("modelId"),
+        "response.model": kwargs.get("modelId"),
+        "vendor": "Bedrock",
+        "ingest_source": "PythonSDK",
+        **compat_fields(["response_time", "duration"], int(time_delta * 1000)),
+        **compat_fields(
+            ["usage.total_tokens", "response.usage.total_tokens"],
+            len(response_body.get("embedding")),
+        ),
```

Comment:
Is this how Bedrock calculates total tokens in their billing for embeddings? It looks like the pricing is based on the input tokens [here](https://aws.amazon.com/bedrock/pricing/)?

Reply:
The API does not support a useful set of tokens at this time. For an AWS Titan LLM model, the input and output tokens are returned as part of the API call. Unfortunately, all we have access to in this scenario is the embedding output vectors which is the best approximation that we have. I actually don't think that length is correct for the tokens, but it is the best we have right now from AWS.

Reply:
After a discussion, pulled the token count. It looks like the resulting embedding may be the accounts total embedding space and not a representation of only the input text.